src/v2.rs

```diff
@@ -357,7 +362,7 @@ impl Aggregator {
         }
 
         let party_ids: Vec<u32> = sig_shares.iter().map(|ss| ss.id).collect();
-        let (Rs, R) = compute::intermediate(msg, &party_ids, nonces);
+        let (Rs, R) = compute::intermediate(msg, self.poly[0], &party_ids, nonces);
```

Comment:
Hmmm, we shouldn't use direct array access here. Same thing in the changes above in this file.

I mean, it should always be fine, but we should add some encapsulation to the type so that we have that guarantee. No need to add encapsulation to this PR though.

Reply:
Also, why can't we use `self.group_key` here?

Reply:
The aggregators don't have a `self.group_key` field.  We could certainly add one, it's just duplicating a single 32-byte value.

`Aggregator::init` does return a `Result`, so we could always check there to make sure that the polynomial isn't empty.  We could also check that each poly is appropriately size for the threshold, that the threshold isn't zero, etc.  Do you want me to add an issue for this so we remember to handle it outside of this PR?